generators/app/templates/specs/mocha/app_spec.js

```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-var */
```

Comment:
`var` will always escape the block / scope in which it is defined in unlike `let` or `const`.

Looking at
https://github.com/eXist-db/generator-exist/blob/72b7ae7483dc0cb295cd6d9553f328a51472f6e2/generators/app/templates/specs/mocha/app_spec.js#L74

the solution could be as simple as defining the variables with `let variable1, variable2` at the very beginning of the function body and then only assign them inside the if-blocks, which open a new scope.
That way you can safely use them later on in the function.
They will be as undefined as they have been with your approach.

Reply:
@line-o  thank you for the suggestion, just to be sure i tested this with a newly generated app.  While following your approach creates an `app_spec.js` that runs without linter warnings, it no longer performs the tests that it currently does. As you say, the values will be undefined, aka always green. The current approach relies on the scope escaping behaviour of `var` so that the value that is read from the files gets passed around. As long as `var` is part of the js language, i d rather spend my time on other features.

Reply:
@duncdrum I understand.
Now that I had a second look, I think it would be best to rethink the tests that rely on `var`.
But since you have more pressing issues we can postpone this discussion to a later point.